bigchaindb/pipelines/block.py

```diff
@@ -58,6 +58,21 @@ def validate_tx(self, tx):
             The transaction (Transaction) if valid, ``None`` otherwise.
         """
         tx = Transaction.from_dict(tx)
+        if self.bigchain.transaction_exists(tx.id):
+            # if the transaction already exists, we must check whether
+            # it's in a valid or undecided block
+            tx, status = self.bigchain.get_transaction(tx.id,
+                                                       include_status=True)
+            if status == self.bigchain.TX_VALID \
+               or status == self.bigchain.TX_UNDECIDED:
+                # if the tx is already in a valid or undecided block,
+                # then it no longer should be in the backlog, or added
+                # to a new block. We can delete and drop it.
+                r.table('backlog').get(tx.id) \
+                        .delete(durability='hard') \
+                        .run(self.bigchain.conn)
+                return None
+
```

Comment:
@rhsimplex You've put all this logic into `Block.validate_tx`.
Would it have made sense to put it into the new Transactions validate method (https://github.com/bigchaindb/bigchaindb/pull/641/files#diff-9f1ab5b5daf1b3072e2123b6939bfb07R13)?